jobbergate-api/jobbergate_api/pagination.py

```diff
@@ -2,18 +2,89 @@
 Pagination feature for all endpoints.
 """
 
+from typing import Generic, List, Optional, TypeVar
 
-class Pagination:
+from pydantic import BaseModel, Field
+from pydantic.generics import GenericModel
+from sqlalchemy import func, select
+
+from jobbergate_api.storage import database
+
+
+class Pagination(BaseModel):
     """
     Basic pagination class.
     """
 
-    def __init__(self, limit: int = 10, skip: int = 0):
+    page: Optional[int] = Field(
+        None,
+        ge=0,
+        description="""
+            The page offset for items.
+            The index for the first item is computed as index == page * per_page.
+            Must be greater than or equal to 0.
```

Comment:
So, this is a zero-based index? I think the description should make this point clear, but I did not come up with a suggestion on how to do it :/

Reply:
OK. I will add some info there.

Reply:
Updated in latest commit.